test/unit/Chainweb/Test/Pact5/RemotePactTest.hs

```diff
@@ -315,116 +349,183 @@ spvTest baseRdb = runResourceT $ do
-        sending v cid clientEnv (NE.singleton cmdParseFailure)
-            & fails ? assertExnContains "Pact parse error"
-
-        cmdInvalidPayloadHash <- do
-            bareCmd <- buildTextCmd v
-                $ set cbChainId cid
-                $ set cbRPC (mkExec "(+ 1 2)" (mkKeySetData "sender00" [sender00]))
-                $ defaultCmd
-            pure $ bareCmd
-                { _cmdHash = Pact5.hash "fakehash"
-                }
-        sending v cid clientEnv (NE.singleton cmdInvalidPayloadHash)
-            & fails ? assertExnContains (validationFailedPrefix cmdInvalidPayloadHash <> "Invalid transaction hash")
-
-        cmdSignersSigsLengthMismatch1 <- do
-            bareCmd <- buildTextCmd v
-                $ set cbSigners [mkEd25519Signer' sender00 []]
-                $ set cbChainId cid
-                $ set cbRPC (mkExec "(+ 1 2)" (mkKeySetData "sender00" [sender00]))
-                $ defaultCmd
-            pure $ bareCmd
-                { _cmdSigs = []
-                }
-        sending v cid clientEnv (NE.singleton cmdSignersSigsLengthMismatch1)
-            & fails ? assertExnContains (validationFailedPrefix cmdSignersSigsLengthMismatch1 <> "Invalid transaction sigs")
-
-        cmdSignersSigsLengthMismatch2 <- liftIO $ do
-            bareCmd <- buildTextCmd v
-                $ set cbSigners []
-                $ set cbChainId cid
-                $ set cbRPC (mkExec "(+ 1 2)" (mkKeySetData "sender00" [sender00]))
-                $ defaultCmd
-            pure $ bareCmd
-                { -- This is an invalid ED25519 signature, but length signers == length signatures is checked first
-                _cmdSigs = [ED25519Sig "fakeSig"]
-                }
-        sending v cid clientEnv (NE.singleton cmdSignersSigsLengthMismatch2)
-            & fails ? assertExnContains (validationFailedPrefix cmdSignersSigsLengthMismatch2 <> "Invalid transaction sigs")
-
-        cmdInvalidUserSig <- liftIO $ do
-            bareCmd <- buildTextCmd v
-                $ set cbSigners [mkEd25519Signer' sender00 []]
-                $ set cbChainId cid
-                $ set cbRPC (mkExec "(+ 1 2)" (mkKeySetData "sender00" [sender00]))
-                $ defaultCmd
-            pure $ bareCmd
-                { _cmdSigs = [ED25519Sig "fakeSig"]
-                }
+                cmdNotEnoughGasFunds <- buildTextCmd v
+                    $ set cbGasPrice (GasPrice 10_000_000_000)
+                    $ set cbGasLimit (GasLimit (Gas 10_000))
+                    $ defaultCmd cid
+                send v cid [cmdNotEnoughGasFunds]
+                    & fails ? P.match _FailureResponse ? P.checkAll
+                        [ P.fun responseStatusCode ? P.equals badRequest400
+                        , P.fun responseBody ? textContains
+                            (validationFailed cmdNotEnoughGasFunds "Attempt to buy gas failed with: BuyGasPactError (PEUserRecoverableError (UserEnforceError \"Insufficient funds\")")
```

Comment:
whoa, why is it returning using `show` instance? It feels like we should pretty print this.

Reply:
Yeah it's a TODO.

Reply:
Not for this PR, though.

Reply:
Can you leave the TODO as a comment in the code, and a ticket linked to it or something?